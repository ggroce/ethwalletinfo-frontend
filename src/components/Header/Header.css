header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(1em + 0.1vw);
}

.header__wallet-button {
  height: 35px;
  border: 1px solid;
  border-radius: 6px;
  border-color: rgb(117, 110, 110);
  width: auto;
  padding-left: 20px;
  padding-right: 20px;
  cursor: pointer;
  font-size: 16px;
  font-weight: bold;
  background-color: #191b20;
  color: rgb(218, 216, 216);
}

.header__wallet-button:hover {
  color: white;
  background-color: #3a404b;
}

.header__go-button {
  height: 30px;
  border: 1px solid;
  border-radius: 6px;
  border-color: rgb(117, 110, 110);
  width: auto;
  padding-left: 10px;
  padding-right: 10px;
  cursor: pointer;
  font-size: 16px;
  font-weight: bold;
  background-color: #191b20;
  color: rgb(218, 216, 216);
  margin-left: 10px;
}

.header__go-button:hover {
  color: white;
  background-color: #3a404b;
}
