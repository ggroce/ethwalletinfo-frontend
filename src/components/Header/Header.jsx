import './Header.css';
import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  resetWallet,
  setAddressStart,
} from '../../redux/wallet/wallet.actions';
import { getTotalGasUsed } from '../../redux/wallet/wallet.selectors';

const Header = () => {
  const [walletConnected, setWalletConnected] = useState(false);
  const [lookupAddress, setLookupAddress] = useState('');
  const dispatch = useDispatch();

  const totalGasUsed = useSelector(getTotalGasUsed);

  const handleFormChange = (event) => {
    const { value } = event.target;
    setLookupAddress(value);
  };

  const submitAddress = (event) => {
    event.preventDefault();
    dispatch(setAddressStart(lookupAddress));
    setWalletConnected(false);
  };

  const connectWallet = async () => {
    try {
      const { ethereum } = window;
      if (!ethereum) {
        alert('MetaMask is not installed');
        return;
      }

      const accounts = await ethereum.request({
        method: 'eth_requestAccounts',
      });
      dispatch(setAddressStart(accounts[0]));
      setWalletConnected(true);
    } catch (error) {
      console.log(error);
    }
  };

  const disconnectWallet = () => {
    dispatch(resetWallet());
    setWalletConnected(false);
  };

  return (
    <header>
      <div>
        {walletConnected === false ? (
          <button className="header__wallet-button" onClick={connectWallet}>
            Connect to Wallet
          </button>
        ) : (
          <button className="header__wallet-button" onClick={disconnectWallet}>
            Disconnect Wallet
          </button>
        )}
        <br />
        OR
        <form onSubmit={submitAddress}>
          <label>Lookup Ethererum Address: </label>
          <input
            type="text"
            maxLength="42"
            onChange={handleFormChange}
            aria-label="address-input"
          />
          <button className="header__go-button" onClick={submitAddress}>
            Lookup
          </button>
        </form>
      </div>
      {totalGasUsed !== null && (
        <h6>
          Total gas spent on transactions: {parseFloat(totalGasUsed).toFixed(8)}{' '}
          Eth
        </h6>
      )}
    </header>
  );
};
export default Header;
