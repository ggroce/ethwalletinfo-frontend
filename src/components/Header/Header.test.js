import { screen, fireEvent } from '@testing-library/react';
import {
  renderWithRedux,
  initialStateWalletStore,
  editWalletStore,
} from '../../test/testing.utils';
import Header from './Header';

test('<Header />: takes a snapshot', () => {
  const { asFragment } = renderWithRedux(<Header />, {
    initialState: initialStateWalletStore,
  });
  expect(
    asFragment(<Header />, {
      initialState: initialStateWalletStore,
    })
  ).toMatchSnapshot();
});

test('<Header />: address input box should update in on input correctly from handleFormChange', () => {
  renderWithRedux(<Header />, { initialState: initialStateWalletStore });

  fireEvent.change(screen.getByLabelText('address-input'), {
    target: { value: '0x123' },
  });
  expect(screen.getByLabelText('address-input')).toHaveValue('0x123');
});

test('<Header />: should display correct gas used value, (0.02 Eth), from supplied redux state', () => {
  const walletStoreWithGasUsed = editWalletStore({ totalGasUsed: 200000 });
  renderWithRedux(<Header />, { initialState: walletStoreWithGasUsed });

  expect(screen.getByRole('heading')).toHaveTextContent(
    'Total gas spent on transactions: 0.02 Eth'
  );
});
