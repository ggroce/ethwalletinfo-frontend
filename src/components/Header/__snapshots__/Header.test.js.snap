// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header />: takes a snapshot 1`] = `
<DocumentFragment>
  <header>
    <div>
      <button
        class="header__wallet-button"
      >
        Connect to Wallet
      </button>
      <br />
      OR
      <form>
        <label>
          Lookup Ethererum Address: 
        </label>
        <input
          aria-label="address-input"
          maxlength="42"
          type="text"
        />
        <button
          class="header__go-button"
        >
          Lookup
        </button>
      </form>
    </div>
  </header>
</DocumentFragment>
`;
